packages/calypso-analytics/src/utils/current-user.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal dependencies
+ */
+import hashPii from './hash-pii';
+
+/**
+ * Module variables
+ */
+let _currentUser: CurrentUser;
+
+export interface CurrentUserHashedPii {
+	ID: string;
+	username: string;
+	email: string;
+}
+
+export interface CurrentUser {
+	ID: number;
+	username: string;
+	email: string;
+	hashedPii?: CurrentUserHashedPii;
+}
+
+/**
+ * Gets current user.
+ *
+ * @returns {object|undefined} Current user.
+ */
+export function getCurrentUser(): CurrentUser {
+	return _currentUser;
+}
+
+/**
+ * Sets current user, (stored in javascript memory).
+ *
+ * @param {object} currentUser the user data for the current user
+ * @returns {object|undefined} Current user.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function setCurrentUser( currentUser: any ): CurrentUser | undefined {
+	if (
+		! currentUser.ID ||
+		isNaN( Number( currentUser.ID ) ) ||
```

Comment:
This logic changed very subtly from what it used to be in JS.

```suggestion
		isNaN( parseInt( currentUser.ID, 10 ) ) ||
```

Without knowing much about the format of the `ID` string, I'd say it's safer to stick with `parseInt` so we can use the explicit radix param.

Reply:
Good catch!